### PR TITLE
controller: change disassociation timer value

### PR DESCRIPTION
**Pull request by CoralMalachi**
Currently, the client steering task sets the access control validity period to disassoc_timer in the client blocking request. However, the disassoc_timer is expressed in TU while according to specification the validity period is expressed in seconds.

In addition, according to section 11.6 in the specification, the Multi-AP agent shall reject
a first attempt by a STA specified in the request to associate to the specified BSS operated by the Multi-AP Agent, only if the following condition is true :

`
The time delta since the message was received is less than the value of the validity period field in the Client Association Control Request message.
`

For that reason, set the validity period to the timeout used by the client steering task
(i.e. steering_wait_time_ms/1000).

#432 
Signed-off-by: Coral Malachi <coral.malachi@intel.com>

### CODE REVIEW

File: controller/src/beerocks/master/tasks/client_steering_task.cpp
Comment (morantr) [inline controller/src/beerocks/master/tasks/client_steering_task.cpp]: Why did you remove it? This line has ended the task.

**Review by morantr — CHANGES_REQUESTED** [review]

File: controller/src/beerocks/master/tasks/client_steering_task.cpp
Comment (CoralMalachi) [reply controller/src/beerocks/master/tasks/client_steering_task.cpp]: you are right I'll fix that.

File: controller/src/beerocks/master/tasks/client_steering_task.cpp
Comment (morantr) [inline controller/src/beerocks/master/tasks/client_steering_task.cpp]: Why did you remove it? please explain.

**Review by morantr — CHANGES_REQUESTED** [review]

File: controller/src/beerocks/master/tasks/client_steering_task.cpp
Comment (CoralMalachi) [reply controller/src/beerocks/master/tasks/client_steering_task.cpp]: In the FINALIZE state, the controller sent an unblock message to all agents, but it's not really needed due to the validity period, so I removed that part. @morantr

Comment (arnout) [reply controller/src/beerocks/master/tasks/client_steering_task.cpp]: > In the FINALIZE state, the controller sent an unblock message to all agents, but it's not really needed due to the validity period, so I removed that part. @morantr

The current bwl implementations don't use the validity period option, so that will have to be implemented first.

I think it's best if you un-remove this code for now, and create a new task to actually use the validity period in bwl. I shouldn't have included that bit in #432 - my bad.

Comment (CoralMalachi) [reply controller/src/beerocks/master/tasks/client_steering_task.cpp]: Ok 👍 @arnout

**Review by arnout — APPROVED** [review]
Small nit about the commit message: the backquote that you used to indicate the text quoted from the specification is not very visually pleasing when reading it as text. It would be better to block-indent the quoted text, or maybe even add `> ` quotation markers like in e-mail.

But I can fix that up while I rebase on master before the merge.

**Comment by CoralMalachi** [comment]
> Small nit about the commit message: the backquote that you used to indicate the text quoted from the specification is not very visually pleasing when reading it as text. It would be better to block-indent the quoted text, or maybe even add `> ` quotation markers like in e-mail.
> 
> But I can fix that up while I rebase on master before the merge.

I'll do rebase now and try what you suggested 👍

**Review by morantr — APPROVED** [review]

File: controller/src/beerocks/master/tasks/client_steering_task.cpp
Comment (CoralMalachi) [reply controller/src/beerocks/master/tasks/client_steering_task.cpp]: #445

**Pull request merged**
